fip-6.md

```diff
@@ -0,0 +1,376 @@
+
+###### Example
+```
+{
+    "receiver_public_key": "FIO8PRe4WRZJj5mkem6qVGKyvNFgPsNnjNN6kPhh6EaCpzCVin5Jj",
+    "votable":0,
+    "unlockable":0,
+    "lock_periods": [ {
+                        “duration”: 86400
+                        “percent: 1.2
+                      }, {
+                        “duration”: 172800
+                        “percent: 90.8
+                      }, {
+                        “duration”: 86400
+                        “percent: 8.0
+                      }
+                    ],
+    "actor": "aftyershcu22",
+    "amount": 40000000000000
+    "max_fee": 40000000000
+}
+```
+##### Processing
+* require auth of the actor
+* Verify the receiver public key.
+* Verify the votable value.
+* Verify the lock periods
+      Minimum of 1
+      check that total is 100%. 
+      check that duration is greater than 0
+* Create account if account does not yet exist for the receiver public key
+*  Verify the locking account has necessary balance.
+* verify that the fee for this does not exceed the max fee specified.
+* charge appropriate fee (this will be a bundled fee transaction, fee will be based on number of periods specified)
+* create entry in the locktokens table for these lock tokens.
+* verify tx does not exceed max transaction size.
+* increase RAM limit by 200+(60*number of lock periods) bytes
+* Return status json containing status (ok), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid receiver public key|Invalid format|400|"receiver_public_key"|value of public key|"Invalid Public Key"|
+|Invalid votable|Invalid format|400|"votable"|votable invalid|"Invalid votable value"|
+|Lock Periods Invalid|less of equal to 0 duration specified on any lock period, or total percent does not sum to 100.0|400|"lock_periods"|Invalid Lock Periods |
+|Invalid Actor|Actor does not exist on chain|403|||"Invalid Actor"|
+|Insufficient balance| Actor does not have required amount of funds|400|"amount"|Value sent in, e.g. "100"|"Insufficient balance"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|max_fee"|Value sent in, e.g. "1000000000"|"Fee exceeds supplied maximum"|
+##### Response
+|Group|Parameter|Format|Definition|
+|---|---|---|---|
+||status|String|Ok|
+||fee_collected|String|fee amount collected SUFs|
+|lock_id| string| the id of the lock that was created
+###### Example
+```
+{
+  "lock_id": 5
+  "status": "Ok",
+  "fee_collected": 0		
+}
+```
+## Fees
+a new fee will be created for lock_tokens, 600000000 SUF per lock period, this is a mandatory fee.
```

Comment:
Shouldn't the fee be based on the duration of the lock? Locking for 1 day (taking RAM for 1 day) costs the same as locking for 100 years (taking RAM for 100 years)?

Reply:
the fee is for the amount of RAM used...im unsure it needs to have this additional time consideration...

Reply:
But used is a combination of size and time. It's not a factor on other calls, because we can purge the data when we deemed it old, e.g. purge request after 90 days. This you can not purge until expiry date and you put no cap on when that can be.